extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/Endpoints.java

```diff
@@ -383,16 +386,28 @@ private static WebSocketEndpoint createEndpoint(String endpointClassName, Contex
             @SuppressWarnings("unchecked")
             Class<? extends WebSocketEndpoint> endpointClazz = (Class<? extends WebSocketEndpoint>) cl
                     .loadClass(endpointClassName);
-            WebSocketEndpoint endpoint = (WebSocketEndpoint) endpointClazz
-                    .getDeclaredConstructor(WebSocketConnectionBase.class, Codecs.class, ContextSupport.class,
-                            SecuritySupport.class)
-                    .newInstance(connection, codecs, contextSupport, securitySupport);
-            return endpoint;
+
+            if (telemetrySupport != null) {
```

Comment:
It's a matter of style but I would prefer something like:
```java
WebSocketEndpoint endpoint = createWebSocketEndpoint(connection, codecs, contextSupport, securitySupport, endpointClazz);
if (telemetrySupport != null) {
   endpoint = telemetrySupport.decorate(endpoint, connection);
}
return endpoint;
```
in order to avoid duplicate `createWebSocketEndpoint()` invocation... sorry for the nitpicking ;-)

Reply:
I prefer that too, usually I'd avoid it because you have one extra variable and I'm not sure that here static analyzer can determine when to avoid that `endpoint` variable. I'll apply that.